### PR TITLE
Additional tests

### PR DESCRIPTION
**Pull request by fang-helen**
Added some tests to cover unchecked edge cases/error handling in UserServlet and Interactions.

### CODE REVIEW

File: src/test/java/com/google/sps/InteractionsTest.java [hunk 1]
```diff
@@ -266,6 +266,47 @@ public void recordMiscInteractions() throws IOException, ServletException {
...
+    datastore.put(firstEntity);
+    datastore.put(secondEntity);
+    // login and view the item
+    PowerMockito.mockStatic(Firebase.class);
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    RequestDispatcher dispatcher = mock(RequestDispatcher.class);
+    when(request.getRequestDispatcher("/WEB-INF/jsp/display-event.jsp")).thenReturn(dispatcher);
+    when(request.getParameter("userToken")).thenReturn(user);
+    PowerMockito.when(Firebase.isUserLoggedIn(anyString())).thenCallRealMethod();
+    PowerMockito.when(Firebase.authenticateUser(anyString())).thenReturn(user);
```
Comment (0tpyrk) [inline src/test/java/com/google/sps/InteractionsTest.java:294]: Could TestingUtils.mockFirebase() be used here instead?

Comment (fang-helen) [reply src/test/java/com/google/sps/InteractionsTest.java:294]: Right, I forgot that was a thing :)
